controllers/secret.go

```diff
@@ -24,187 +24,109 @@ import (
 	if err != nil {
-		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+		log.Error(err, "unable to resolve github secret for %s/%s", meta.Namespace, meta.Name)
 		return err
 	}
-	if kubeConfigSecret.Data[KUBECONFIG_SECRET] == nil {
-		err = errors.New(fmt.Sprintf("the kubeconfig key in %s is empty!", secretName))
-		log.Error(err, fmt.Sprintf("The %s secret does not contain a kubeconfig", secretName))
-		return err
+	secretName := defaultGithubApiSecretName
+	if operatorPipeline.Spec.GitHubSecretName != "" {
+		secretName = operatorPipeline.Spec.GitHubSecretName
 	}
-	return nil // Existing Secret found, do nothing...
+	return r.ensureSecret(secretName, defaultGithubApiSecretKeyName, meta)
 }
 
-// reconcileGitHubAPISecret will ensure that the GitHub API Secret is present and up to date.
-func (r *OperatorPipelineReconciler) reconcileGitHubAPISecret(meta metav1.ObjectMeta) error {
-
-	namespaceName := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      meta.Name,
-	}
-
-	operatorPipeline := &certv1alpha1.OperatorPipeline{}
-	err := r.Client.Get(context.TODO(), namespaceName, operatorPipeline)
+// ensurePyxisAPISecret will ensure that the Pyxis API Secret is present and up to date.
+func (r *OperatorPipelineReconciler) ensurePyxisAPISecret(meta metav1.ObjectMeta) error {
+	operatorPipeline, err := r.getPipeline(meta)
 	if err != nil {
-		if k8errors.IsNotFound(err) {
-			log.Info("pipeline resource not found. Ignoring since object must be deleted")
-			return nil
-		}
-
-		log.Error(err, fmt.Sprintf("unable to retrieve pipeline resource in namespace: %s ", meta.Namespace))
+		log.Error(err, "unable to resolve pyxis secret for %s in %s", meta.Name, meta.Namespace)
 		return err
 	}
-
-	var gitHubSecretName = GITHUB_API_SECRET
-	if operatorPipeline.Spec.GitHubSecretName != "" {
-		gitHubSecretName = operatorPipeline.Spec.GitHubSecretName
-	}
-
-	key := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      gitHubSecretName,
+	secretName := defaultPyxisApiSecretName
+	if operatorPipeline.Spec.PyxisSecretName != "" {
+		secretName = operatorPipeline.Spec.PyxisSecretName
 	}
+	return r.ensureSecret(secretName, defaultPyxisApiSecretKeyName, meta)
+}
 
-	secret := newSecret(key)
-	if !IsObjectFound(r.Client, key, secret) {
-		return errors.New(fmt.Sprintf("github api secret not found in namespace: %s", meta.Namespace))
+// ensureSecret will ensure that the a secret with the appropriate name and key name are present
+func (r *OperatorPipelineReconciler) ensureSecret(secretName string, secretKeyName string, meta metav1.ObjectMeta) error {
+	namespacedSecretName := newNamespacedName(secretName, meta.Namespace)
+	secret := &corev1.Secret{}
+	if !IsObjectFound(r.Client, namespacedSecretName, secret) {
+		err := errors.New(fmt.Sprintf("could not find existing secret %s/%s", meta.Namespace, secretName))
+		log.Error(err, fmt.Sprintf("unable to ensure secret %s/%s", meta.Namespace, secretName))
```

Comment:
Same here, error wrapping could be used by defining a generic error (e.g. `unable to ensure secret`) in a file and add further context with `fmt.Sprintf("could not find existing secret %s/%s", meta.Namespace, secretName)`.